components/service/glean/src/test/java/mozilla/components/service/glean/GleanTest.kt

```diff
@@ -283,28 +249,74 @@ class GleanTest {
+
+        // Define a 'click' event for the purpose of validating events ping schema
+        val click = EventMetricType(
+            disabled = false,
+            category = "ui",
+            lifetime = Lifetime.Ping,
+            name = "click",
+            sendInPings = listOf("events"),
+            objects = listOf("buttonA", "buttonB")
+        )
+        // Record two events of the same type, with a little delay.
+        click.record("buttonA")
+        val expectedTimeSinceStart: Long = 37
+        SystemClock.sleep(expectedTimeSinceStart)
+        click.record("buttonB")
 
-            val metricsJson = JSONObject(requests["metrics"])
-            checkPingSchema(metricsJson)
+        // Define a StringMetricType for validation of the metrics ping schema
+        val stringMetric = StringMetricType(
+            disabled = false,
+            category = "test",
+            lifetime = Lifetime.Ping,
+            name = "test_string",
+            sendInPings = listOf("metrics")
+        )
+        stringMetric.set("Glean Rules!")
 
-            // Check ping type
-            val pingInfo = metricsJson.getJSONObject("ping_info")!!
-            assertEquals(pingInfo["ping_type"], "metrics")
+        // Fake calling the lifecycle observer.
+        val lifecycleRegistry = LifecycleRegistry(Mockito.mock(LifecycleOwner::class.java))
+        val context: Context = ApplicationProvider.getApplicationContext()
+        val pingScheduler = PingScheduler(context,
+            StorageEngineManager(applicationContext = context))
+        lifecycleRegistry.addObserver(pingScheduler)
+        try {
+            // Simulate the first foreground event after the application starts.
+            lifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_RESUME)
+
+            // Simulate going to background.
+            lifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_STOP)
 
-            // Retrieve the string metric that was stored
-            val metricsObject = metricsJson.getJSONObject("metrics")!!
-            val stringMetrics = metricsObject.getJSONObject("string")!!
+            // Make sure the PingUploadWorker is scheduled
+            assertTrue(PingScheduler.isWorkScheduled())
 
-            assertEquals(stringMetrics["telemetry.metrics_ping"], "Test metrics ping")
+            val pingList = pingScheduler.pingStorageEngine.storageDirectory.listFiles()
+
+            // This should at contain a baseline, events, and a metrics ping
+            pingList?.forEach { file ->
+                val bufferedReader = BufferedReader(FileReader(file))
+                val path = bufferedReader.readLine()
+                val serializedPing = bufferedReader.readLine()
+
+                // Compare the path UUID to the filename
+                assertEquals(path.substring(path.lastIndexOf("/") + 1), file.name)
+                // Validate the ping schema
+                checkPingSchema(serializedPing)
+            }
```

Comment:
Awesome.  Thanks for providing a way to check ping contents.  I wonder if we should provide a test helper function to handle the loading of contents (i.e. the first three lines in this block).

(Also, I liked it better before when the `PingStorageEngine` abstracted away how to get the individual files, but that's *extreme* bikeshedding).